src/box/relay.cc

```diff
@@ -188,6 +188,14 @@ struct relay {
 	double txn_lag;
 	/** Relay sync state. */
 	enum relay_state state;
+	/** Whether relay should speed up the next heartbeat dispatch. */
+	bool need_new_vclock_sync;
+	/**
+	 * Whether relay is in process of sending a tx to the remote peer. In
+	 * this case cannot interrupt the transaction stream with any heartbeats
```

Comment:
Why?

Reply:
Because applier reads row stream in transactions. A ping (which's not DML, it's IPROTO_OK, not IPROTO_NOP) will break the transaction.
See here, for example:
https://github.com/tarantool/tarantool/blob/dc147ec9190fc8bbbe56ce8a4a1718fbf8b03484/src/box/applier.cc#L831-L836
It's not that easy to rework that. And I don't see why we should.

Reply:
Okay, let's leave as is. It doesn't seem to be critical.